api/v1beta1/keystoneapi.go

```diff
@@ -69,11 +70,33 @@ func GetKeystoneAPI(
 	return &keystoneList.Items[0], nil
 }
 
-// GetAdminServiceClient - get an admin serviceClient for the keystoneAPI instance
+// GetAdminServiceClient - get a system scoped admin serviceClient for the keystoneAPI instance
 func GetAdminServiceClient(
 	ctx context.Context,
 	h *helper.Helper,
 	keystoneAPI *KeystoneAPI,
+) (*openstack.OpenStack, ctrl.Result, error) {
+	os, ctrlResult, err := GetScopedAdminServiceClient(
+		ctx,
+		h,
+		keystoneAPI,
+		&gophercloud.AuthScope{
+			System: true,
+		},
+	)
+	if err != nil {
+		return nil, ctrlResult, err
+	}
+
+	return os, ctrlResult, nil
+}
+
+// GetScopedAdminServiceClient - get a scoped admin serviceClient for the keystoneAPI instance
+func GetScopedAdminServiceClient(
+	ctx context.Context,
+	h *helper.Helper,
+	keystoneAPI *KeystoneAPI,
+	scope *gophercloud.AuthScope,
 ) (*openstack.OpenStack, ctrl.Result, error) {
 	// get public endpoint as authurl from keystone instance
 	authURL, err := keystoneAPI.GetEndpoint(endpoint.EndpointInternal)
```

Comment:
The following parameters:

```
Username:   keystoneAPI.Spec.AdminUser,
TenantName: keystoneAPI.Spec.AdminProject,
```

are not needed here to choose the admin user in the admin  project?

Reply:
not sure I understand, the AdminUser and Project is used L138/L140. you refer to that one?

Reply:
oh yes, it was outside of the changes so i didnt realized it was already being done.